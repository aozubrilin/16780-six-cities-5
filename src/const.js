export const STAR_TO_PERSENT = 20;

export const OfferCardClass = {
  MAIN: `cities`,
  NEAR: `near-places`,
  FAVORITE: `favorites`,
};

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const SortType = {
  POPULAR: `Popular`,
  TO_HIGHT: `Price: low to high`,
  TO_LOW: `Price: high to low`,
  TOP_RATE: `Top rated first`,
};
