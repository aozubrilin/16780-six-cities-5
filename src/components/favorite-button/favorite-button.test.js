import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";
import FavoriteButton from "./favorite-button";
import {favoriteButtonProperty} from "../../const";


const noop = () => {};
const testStore = configureStore();

describe(`FavoriteButton snapshot`, () => {
  describe(`with authorizationStatus: AUTH`, () => {

    const store = testStore({
      user: {
        authorizationStatus: `AUTH`,
      }
    });

    it(`FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.offerCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={false}
                buttonProperty={favoriteButtonProperty.offerCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.offerCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={true}
                buttonProperty={favoriteButtonProperty.offerCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.propertyCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={false}
                buttonProperty={favoriteButtonProperty.propertyCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.propertyCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={true}
                buttonProperty={favoriteButtonProperty.propertyCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`with authorizationStatus: NO_AUTH`, () => {

    const store = testStore({
      user: {
        authorizationStatus: `NO_AUTH`,
      }
    });

    it(`FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.offerCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={false}
                buttonProperty={favoriteButtonProperty.offerCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.offerCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={true}
                buttonProperty={favoriteButtonProperty.offerCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.propertyCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={false}
                buttonProperty={favoriteButtonProperty.propertyCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.propertyCard}`, () => {
      const tree = renderer.create(
          <Provider store={store}>
            <MemoryRouter>
              <FavoriteButton
                offerId={1}
                isFavorite={true}
                buttonProperty={favoriteButtonProperty.propertyCard}
                onChangeFavoriteSatus={noop}
              />
            </MemoryRouter>
          </Provider>
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

});
