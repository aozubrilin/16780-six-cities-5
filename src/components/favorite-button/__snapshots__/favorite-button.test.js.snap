// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton snapshot with authorizationStatus: AUTH FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.offerCard} 1`] = `
<button
  className="place-card__bookmark-button button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To
     bookmarks
  </span>
</button>
`;

exports[`FavoriteButton snapshot with authorizationStatus: AUTH FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.propertyCard} 1`] = `
<button
  className="property__bookmark-button button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To
     bookmarks
  </span>
</button>
`;

exports[`FavoriteButton snapshot with authorizationStatus: AUTH FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.offerCard} 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>
</button>
`;

exports[`FavoriteButton snapshot with authorizationStatus: AUTH FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.propertyCard} 1`] = `
<button
  className="property__bookmark-button property__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>
</button>
`;

exports[`FavoriteButton snapshot with authorizationStatus: NO_AUTH FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.offerCard} 1`] = `
<a
  className="place-card__bookmark-button button"
  href="/login"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To
     bookmarks
  </span>
</a>
`;

exports[`FavoriteButton snapshot with authorizationStatus: NO_AUTH FavoriteButton render correctly with isFavorite={false} and buttonProperty={favoriteButtonProperty.propertyCard} 1`] = `
<a
  className="property__bookmark-button button"
  href="/login"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To
     bookmarks
  </span>
</a>
`;

exports[`FavoriteButton snapshot with authorizationStatus: NO_AUTH FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.offerCard} 1`] = `
<a
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  href="/login"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>
</a>
`;

exports[`FavoriteButton snapshot with authorizationStatus: NO_AUTH FavoriteButton render correctly with isFavorite={true} and buttonProperty={favoriteButtonProperty.propertyCard} 1`] = `
<a
  className="property__bookmark-button property__bookmark-button--active button"
  href="/login"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In
     bookmarks
  </span>
</a>
`;
